### PR TITLE
PLAT-777 | Laravel Doc

### PR DESCRIPTION
**Pull request by kvas-damian**
@gitkuba can you recommend any sources from Laracasts?

### CODE REVIEW

File: laravel.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+* I can set data in users session
+* I can introduce Authorization to my application - only logged in users can see the content
+* I can validate data provided by user to make sure the content is in the right shape
+* I can limit permissions to introduce admin role
+
+### Advanced aka _My TODO app has REST API_
+**Prerequisite knowledge:** rest, http,
+* I can read headers from request and set headers on a response
+* I can create REST API in my app
+* I can use Lavarel transformers to make the output consistent
```
Comment (kvas-damian) [inline laravel.md]: What do you think about adding integration with Vue here?
I personally have a mixed feeling about putting it in docs about the Laravel, but it's very common to follow that path.

File: laravel.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+* I can use Jobs, Queues and Scheduler for maintenance or writing asynchronous code
+* I can test my code using Unit Tests and Acceptance Tests
+* I can deploy my code to production
+
+## Sources
+
+### Books
+- [Laravel: Up & Running: A Framework for Building Modern PHP Apps](https://laravelupandrunning.com/)
+
+### Read online
```
Comment (bognix) [inline laravel.md:48]: I can recommend: https://laracasts.com/series/laravel-from-scratch-2018

File: laravel.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
+# Laravel
+
+**Prerequisite knowledge:** php, oop, mysql, html, frameworks, mvc
```
Comment (bognix) [inline laravel.md]: I would add composer here although on the other hand you can use laravel using the homestead and then everything is set up for you.

Comment (bognix) [reply laravel.md]: In one of those levels I would mention dependency management anyway.

File: laravel.md (hunk 1 above)
Comment (bognix) [reply laravel.md]: I'm not sure about the vue but I would mention assets somewhere for sure.

File: laravel.md (hunk 2 above)
Comment (kvas-damian) [reply laravel.md:48]: Noted, thanks

File: laravel.md (hunk 1 above)
Comment (kvas-damian) [reply laravel.md]: Good Idea, I put it in `Intermediate` level

File: laravel.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# Laravel
+
+**Prerequisite knowledge:** php, oop, mysql, html, frameworks, mvc, composer, dependency managment
```
Comment (gitkuba) [inline laravel.md:3]: I've learned oop, mvc, frameworks and composer as I was making my first project in Symfony 2. And it seems natural to me that you explore those concepts as you explore your first MVC framework. Therefore I'm not sure they should be classified as prerequisite knowledge... It's rather a digress than suggestion though. 😄

File: laravel.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+## Intro
+Laravel is a free, open-source PHP web framework, created by Taylor Otwell and intended for the development of web applications following the model–view–controller (MVC) architectural pattern and based on Symfony. Some of the features of Laravel are a modular packaging system with a dedicated dependency manager, different ways for accessing relational databases, utilities that aid in application deployment and maintenance, and its orientation toward syntactic sugar. [[Source](https://en.wikipedia.org/wiki/Laravel)]
+
+## Classification / acceptance questions
+
+### Apprentice aka _My First TODO app_
+* I know what Laravel is
```
Comment (rogatty) [inline laravel.md]: Is it enough to read the intro?

File: laravel.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+* I can use Laravel router to interact with the end user
```
Comment (rogatty) [inline laravel.md]:
```suggestion
* I can use Laravel router to respond to client requests
```

File: laravel.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+* I can create simple blade templates to separate presentation layer from business logic
+* I can Create, Read, Update, Delete entries in DB through Eloquent Model 
+* I can use MVC design pattern in Laravel
```
Comment (rogatty) [inline laravel.md]: This sounds very generic, can you add some examples or split it into more specific points?

File: laravel.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+## Intro
+Laravel is a free, open-source PHP web framework, created by Taylor Otwell and intended for the development of web applications following the model–view–controller (MVC) architectural pattern and based on Symfony. Some of the features of Laravel are a modular packaging system with a dedicated dependency manager, different ways for accessing relational databases, utilities that aid in application deployment and maintenance, and its orientation toward syntactic sugar. [[Source](https://en.wikipedia.org/wiki/Laravel)]
+
+## Classification / acceptance questions
+
+### Apprentice aka _My First TODO app_
+* I know what Laravel is
+* I can use Laravel router to interact with the end user
+* I can create simple blade templates to separate presentation layer from business logic
+* I can Create, Read, Update, Delete entries in DB through Eloquent Model
```
Comment (rogatty) [inline laravel.md]:
```suggestion
* I can Create, Read, Update and Delete entries in DB through Eloquent Model 
```

File: laravel.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+### Apprentice aka _My First TODO app_
+* I know what Laravel is
+* I can use Laravel router to interact with the end user
+* I can create simple blade templates to separate presentation layer from business logic
+* I can Create, Read, Update, Delete entries in DB through Eloquent Model 
+* I can use MVC design pattern in Laravel
+
+### Intermediate aka _Beta users can use my TODO app_
+**Prerequisite knowledge:** cookies, session, authorization
+* I can use artisian CLI for DB migration and code generation
```
Comment (rogatty) [inline laravel.md]:
```suggestion
* I can use Artisan CLI for DB migration and code generation
```

File: laravel.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+* I can set data in users session
+* I can introduce Authorization to my application - only logged in users can see the content
+* I can set Eloquest relations to fetch related data from DB
```
Comment (rogatty) [inline laravel.md]:
```suggestion
* I can set Eloquent relations to fetch related data from DB
```

File: laravel.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+* I can use Laravel router to interact with the end user
+* I can create simple blade templates to separate presentation layer from business logic
+* I can Create, Read, Update, Delete entries in DB through Eloquent Model 
+* I can use MVC design pattern in Laravel
+
+### Intermediate aka _Beta users can use my TODO app_
+**Prerequisite knowledge:** cookies, session, authorization
+* I can use artisian CLI for DB migration and code generation
+* I can set data in users session
+* I can introduce Authorization to my application - only logged in users can see the content
```
Comment (rogatty) [inline laravel.md]: Should we also mention authentication?

Comment (rogatty) [inline laravel.md]: This is pretty generic, should we specify policies and gates? https://laravel.com/docs/5.7/authorization#introduction

File: laravel.md (hunk 7 above)
Comment (kvas-damian) [reply laravel.md]: This is a summary of 3 points above. I'm not sure if I can phrase an example here that will be understood by Aprentice. Maybe it's better to remove it?
On the other hand I think it's important for a person that starts his journey with frameworks to know that MVC is important part of his life.

File: laravel.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+* I can Create, Read, Update and Delete entries in DB through Eloquent Model 
+* I can use MVC design pattern in Laravel
+
+### Intermediate aka _Beta users can use my TODO app_
+**Prerequisite knowledge:** cookies, session, authorization
+* I can use Artisan CLI for DB migration and code generation
+* I can set data in users session
+* I can introduce Authorization to my application - only logged in users can see the content
+* I can set Eloquent relations to fetch related data from DB
+* I can validate data provided by user to make sure the content is in the right shape
```
Comment (rogatty) [inline laravel.md]:
```suggestion
* I can validate user input to make sure the incoming data is in the expected shape
```

File: laravel.md (hunk 4 above)
Comment (kvas-damian) [reply laravel.md:3]: I think it depends on the sources that you use. Many books that are introducing you to a framework starts with explaining many of these areas, but I saw some articles that skips it. I think it's better to point it in two places than skipping.

File: laravel.md [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+* I can limit permissions to introduce admin role
```
Comment (rogatty) [inline laravel.md]:
```suggestion
* I can assign roles with a specific permissions set to users
```

File: laravel.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+* I can serve static assets like css and js to make user experience better
```
Comment (rogatty) [inline laravel.md]:
```suggestion
* I can serve static assets like CSS and JS to make user experience better
```

File: laravel.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+* I can use Artisan CLI for DB migration and code generation
+* I can set data in users session
+* I can introduce Authorization to my application - only logged in users can see the content
+* I can set Eloquent relations to fetch related data from DB
+* I can validate user input to make sure the incoming data is in the expected shape
+* I can assign roles with a specific permissions set to users
+* I can serve static assets like CSS and JS to make user experience better
+
+### Advanced aka _My TODO app has REST API_
+**Prerequisite knowledge:** rest, http,
```
Comment (rogatty) [inline laravel.md]:
```suggestion
**Prerequisite knowledge:** rest, http
```

File: laravel.md [hunk 16]
```diff
@@ -0,0 +1,52 @@
...
+* I can read headers from request and set headers on a response
```
Comment (rogatty) [inline laravel.md]:
```suggestion
* I can read headers from a request and set headers on a response
```

File: laravel.md [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+* I can create REST API in my app
+* I can use Lavarel transformers to make the output consistent
```
Comment (rogatty) [inline laravel.md]:
```suggestion
* I can use Laravel transformers to make the output consistent
```

File: laravel.md (hunk 4 above)
Comment (rogatty) [reply laravel.md:3]: I think our dependencies system isn't really about "prerequisite knowledge" but more about the things you'll also learn in the process.

Comment (kvas-damian) [reply laravel.md:3]: We can start a discussion in a separate issue or PR.

File: laravel.md (hunk 5 above)
Comment (kvas-damian) [reply laravel.md]: Do you have any suggestions here?

Comment (rogatty) [reply laravel.md]: I'd remove it. You already know what Laravel is at this point of doc.

File: laravel.md (hunk 7 above)
Comment (rogatty) [reply laravel.md]: How about this?
```
* I can Create, Read, Update and Delete entries in DB through Eloquent Model (M in MVC)
* I can create simple blade templates to separate presentation layer from business logic (V in MVC)
* I can use Laravel router to respond to client requests (C in MVC)
```

**Review by rogatty — APPROVED** [review]

**Pull request merged**